### ISSUE TITLE
Inline values provider isn't called with new view port when scrolling editor content

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode#119559]
from https://github.com/microsoft/vscode/issues/105690#issuecomment-804710058

`DebugEditorContribution.updateInlineValueDecorations` needs only be called when a provider is registered. The built-in fallback solution does not need to be called on scrolling because it does not try to only show inline values for the visible area.

**Comment by weinand** [comment]
@isidorn thanks, the fix works great.

Initially I thought that the viewport passed to the inline provider is a bit off, but then I noticed that I'm passing `getVisibleRangesPlusViewportAboveBelow()` which adds more lines above and below which is a good idea as my experiments show:

With `getVisibleRanges()` new lines appear without inline values and then after 200ms the values appear.

![2021-03-29_16-38-34 (1)](https://user-images.githubusercontent.com/1898161/112862811-72d5e480-90b6-11eb-97a2-0cedd78d0ebc.gif)

With `getVisibleRangesPlusViewportAboveBelow()` the inline values scroll smoothly into the view because they were already added in the "PlusViewportAboveBelow" area.

![2021-03-29_16-44-29 (1)](https://user-images.githubusercontent.com/1898161/112862850-7d907980-90b6-11eb-849b-c2c6883e146e.gif)

Of course that only helps if scrolling stays within the "PlusViewportAboveBelow" area. For page wise scrolling it won't help.

**Comment by isidorn** [comment]
@weinand thanks for the gifs, they show the two strategies quite well.
Based on your comment I am adding verfied label.

**Comment by isidorn** [comment]
@testforstephen please note that we debounce this call. So we only call the provider once the user stops scrolling.
I think in practice there will not be too many calls. Did you have a chance to try it out?

**Comment by testforstephen** [comment]
Yes, I tried it in Java debugger and debounce is taking effect. It's not the debounce I'm worried about, it's the fact that the inline values returned are volatile and not cached. As I scrolled up and down in the editor, new requests kept coming and I had to recalculate the same lines again.

Here is an example. In the very first window, the inline values in the stack frame are fully displayed, then scrolling down and back, the previous results are lost and I have to recalculate the results again. Ideally, it should be like lazy loading. I would prefer not to refresh the view again if the results have been provided before.

https://user-images.githubusercontent.com/14052197/113988755-5ab65000-9882-11eb-8a1f-b4665effe355.mov
